prime-router/docs/openapi.yml

```diff
@@ -9,6 +9,380 @@ info:
   version: 0.2.0-oas3
 paths:
 
+  # The sender end-point handles receive actions.
+  #
+  # END-POINTS
+  # /submissions
+  # /submissions/multorg
+  # /submissions/{reportid}
+  # /submissions/{organizationName}
+  #
+  # FILTERS (OPTIONAL)
+  # ?offset=INT
+  # ?limit=INT
+  # ?dateStart=STRING (YYYY-MM-DD) || INT (NN [days back] | default: 30)
+  # ?dateEnd=STRING (YYYY-MM-DD | default: Today)
+  # ?year=INT (YYYY | default: ALL)
+  # ?month=INT (MM | default: ALL | requires ?year)
+  # ?organizations=INT (default: ALL)
+  # ?recipients=ARRAY
+  # ?sender=ARRAY
+  # ?warnings=BOOLEAN
+  # ?errors=BOOLEAN
+  # ?receipts=BOOLEAN
+  #
+  #
+  #
+  #
+  # 30 days (based on where request orignates)
+  # pagination?
```

Comment:
Think we can do pagination using `offset` and `limit` (aka pagesize) in the same call?

Reply:
How would you want to handle that? Like `/?pagesize=N`? Or would you want to set the page size as a setting and then do `/endpoint/1/`, /endpoint/2/`, /endpoint/{PAGENUMBER}/`

Reply:
Good question. I'm not sure what is considered standard, so I don't have a strong opinion on the exact syntax. Jim Duff's opinion is probably valuable here.

I would like to double-check that the client can pass **both** `offset` and `pagesize` in the same call. 

When doing pagination, the client UI typically looks like this:
![image](https://user-images.githubusercontent.com/74203452/135675536-6ab38991-42d0-445d-9af8-38ed4bd0a87c.png)

When the user clicks on Page 5, it would call the API passing `offset=5` **and** `pagesize=25`. The server then does the math of 25*5 to get the actual row start offset, then does a query limit of 25 to only return the data that's going to be displayed.

Reply:
Ah gotcha. Yeah so as is you could do `/endpoint?offset=N&limit=25&anotherQuery=N` we could rename it /endpoint?page=N&pagesize=N`. That way `page * pagesize = offset` is really what you're doing. `offset` generally indicates how many items you skip. But yeah, definitely can chain queries.

Reply:
Good point about offset (as an absolute) vs page (which is dependant on the pagesize).

I think Java Spring uses "page" and "size" [see this article - but have an ad blocker](https://www.baeldung.com/rest-api-pagination-in-spring). 
But the client can also do the math and give an absolute "offset" and "limit". 

Defining _which_ approach is being used is probably the important part. 

Interested to hear what others think.